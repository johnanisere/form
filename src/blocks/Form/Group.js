import styled from "styled-components";
import Section from "../../elements/Section";

const Group = styled(Section)`
  flex-direction: column;
  padding: 20px;
`;

export default Group;
