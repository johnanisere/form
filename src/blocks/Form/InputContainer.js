import styled, { css } from "styled-components";
import Article from "../../elements/Article";

const InputContainer = styled(Article)`
  ${({ special }) =>
    special &&
    css`
      display: flex;
      justify-content: center;
      padding: 20px;
      input {
        font-size: 14px;
        color: rgb(255, 255, 255);
        outline: none;
        background: linear-gradient(rgb(99, 113, 199), rgb(85, 99, 193));
        &:hover ,&:focus{
          box-shadow: 2px 4px 11px 1px rgba(0,0,0,.2)
          transform: translateY(-1);
        }
      }
    `}
`;

export default InputContainer;
