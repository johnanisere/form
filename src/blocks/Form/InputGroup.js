import styled, { css } from "styled-components";
import P from "../../elements/P";

const InputGroup = styled(P)`
  ${({ special }) =>
    special &&
    css`
      display: flex;
      justify-content: center;
      padding: 20px;
      input {
        font-size: 14px;
        color: rgb(255, 255, 255);
        outline: none;
        background: linear-gradient(rgb(99, 113, 199), rgb(85, 99, 193));
        cursor: pointer;
      }
    `}
`;

export default InputGroup;
