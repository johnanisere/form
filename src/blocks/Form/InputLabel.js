import styled from "styled-components";
import Label from "../../elements/Label";

const InputLabel = styled(Label)`
  display: block;
  width: 100%;
  box-sizing: border-box;
  padding-bottom: 6px;
  color: #212b36;
  font-size: 14px;
`;

export default InputLabel;
