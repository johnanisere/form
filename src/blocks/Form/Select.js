import styled from "styled-components";
import MySelect from "../../elements/Select";

const Select = styled(MySelect)`
  width: 100%;
  display: block;
  box-sizing: border-box;
  border: 1px solid #e3e5e7;
  height: 120px;
  font-size: 14px;
  line-height: 1.42857143;
  padding: 18px;
  color: #555;
  background-color: #fff;
  background-image: none;
  border: 1px solid #ccc;
  border-radius: 4px;
  box-shadow: inset 0 1px 1px rgba(0, 0, 0, 0.075);
  transition: border-color ease-in-out 0.15s, box-shadow ease-in-out 0.15s;
  &:focus {
    border-width: 2px !important;
    border-style: solid !important;
    border-color: rgb(92, 106, 196) !important;
    border-image: initial !important;
    outline: white;
  }
`;

export default Select;
