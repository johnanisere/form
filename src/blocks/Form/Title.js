import styled from "styled-components";
import H1 from "../../elements/H1";

const Title = styled(H1)`
  margin: 0;
  padding: 20px;
  text-align: left;
  border-bottom: 1px solid rgb(235, 238, 240);
  margin-bottom: 20px;
  font-size: 24px;
  font-weight: 600;
  padding: 20px;
  color: #212b36;
`;

export default Title;
