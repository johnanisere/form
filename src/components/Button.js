import React from "react";

import Form from "../blocks/Form";

const Button = ({ fetching }) => {
  const value = fetching ? "Processing..." : "Submit";
  return (
    <Form.InputGroup special={true}>
      <Form.Input type="submit" disabled={fetching} value={value} />
    </Form.InputGroup>
  );
};

export default Button;
