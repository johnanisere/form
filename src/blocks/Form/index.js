import styled from "styled-components";

import Input from "./Input";
import Group from "./Group";
import Title from "./Title";
import Select from "./Select";
import InputGroup from "./InputGroup";
import InputLabel from "./InputLabel";
import MyForm from "../../elements/Form";
import InputContainer from "./InputContainer";

const Form = styled(MyForm)`
  width: 100%;
  max-width: 580px;
  box-shadow: rgba(63, 63, 68, 0.05) 0px 0px 0px 1px,
    rgba(63, 63, 68, 0.15) 0px 1px 3px 0px;
  border-radius: 3px;
  margin: 50px 0;

  .geosuggest__input {
    width: 100%;
    display: block;
    box-sizing: border-box;
    padding: 4px 18px;
    border: 1px solid #e3e5e7;
    height: 40px;
    font-size: 14px;
    line-height: 1.42857143;
    color: #555;
    background-color: #fff;
    background-image: none;
    border: 1px solid #ccc;
    border-radius: 4px;
    box-shadow: inset 0 1px 1px rgba(0, 0, 0, 0.075);
    transition: border-color ease-in-out 0.15s, box-shadow ease-in-out 0.15s;
    &:focus {
      border-width: 2px !important;
      border-style: solid !important;
      border-color: rgb(92, 106, 196) !important;
      border-image: initial !important;
      outline: white;
    }
  }
  .geosuggest__suggests {
    list-style: none;
    padding: 0;
    background: #ffff;
    margin: 0;
    box-shadow: rgba(63, 63, 68, 0.05) 0px 0px 0px 1px,
      rgba(63, 63, 68, 0.15) 0px 1px 3px 0px;
    border-radius: 3px;
    li {
      padding: 0.5em 20px;
      cursor: pointer;
      &:hover {
        background: #f5f5f5;
      }
    }
  }
  .geosuggest__suggests--hidden {
    display: none;
  }
`;

Form.Title = Title;
Form.Input = Input;
Form.Group = Group;
Form.Select = Select;
Form.Label = InputLabel;
Form.InputGroup = InputGroup;
Form.InputContainer = InputContainer;

export default Form;
