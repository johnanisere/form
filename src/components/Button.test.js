import React from "react";
import { shallow } from "enzyme";

import Button from "./Button";
import Form from "../blocks/Form";

it("Button component exists", () => {
  const wrapper = shallow(<Button />);
  expect(wrapper.exists()).toBe(true);
});

it("Button component renders", () => {
  const wrapper = shallow(<Button />);
  expect(wrapper).toMatchSnapshot();
});

it(`Button should contain "Form.Input"`, () => {
  const props = {
    fetching: false
  };
  const wrapper = shallow(<Button {...props} />);
  expect(wrapper.find(Form.Input)).toHaveLength(1);
});
