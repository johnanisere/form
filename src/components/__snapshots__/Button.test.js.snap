// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <ForwardRef
        type="submit"
        value="Submit"
      />,
      "special": true,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "disabled": undefined,
        "type": "submit",
        "value": "Submit",
      },
      "ref": null,
      "rendered": null,
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bwzfXH",
          "isStatic": true,
          "rules": Array [
            "
  width: 100%;
  display: block;
  box-sizing: border-box;
  padding: 4px 18px;
  border: 1px solid #e3e5e7;
  height: 40px;
  font-size: 14px;
  line-height: 1.42857143;
  color: #555;
  background-color: #fff;
  background-image: none;
  border: 1px solid #ccc;
  border-radius: 4px;
  box-shadow: inset 0 1px 1px rgba(0, 0, 0, 0.075);
  transition: border-color ease-in-out 0.15s, box-shadow ease-in-out 0.15s;
  &:focus {
    border-width: 2px !important;
    border-style: solid !important;
    border-color: rgb(92, 106, 196) !important;
    border-image: initial !important;
    outline: white;
  }
",
          ],
        },
        "displayName": "Styled(styled.input)",
        "foldedComponentIds": Array [
          "sc-bdVaJa",
        ],
        "render": [Function],
        "styledComponentId": "sc-bwzfXH",
        "target": "input",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-dnqmqq",
        "isStatic": false,
        "rules": Array [
          "
  ",
          [Function],
          "
",
        ],
      },
      "displayName": "Styled(styled.p)",
      "foldedComponentIds": Array [
        "sc-htoDjs",
      ],
      "render": [Function],
      "styledComponentId": "sc-dnqmqq",
      "target": "p",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <ForwardRef
          type="submit"
          value="Submit"
        />,
        "special": true,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "disabled": undefined,
          "type": "submit",
          "value": "Submit",
        },
        "ref": null,
        "rendered": null,
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": true,
            "rules": Array [
              "
  width: 100%;
  display: block;
  box-sizing: border-box;
  padding: 4px 18px;
  border: 1px solid #e3e5e7;
  height: 40px;
  font-size: 14px;
  line-height: 1.42857143;
  color: #555;
  background-color: #fff;
  background-image: none;
  border: 1px solid #ccc;
  border-radius: 4px;
  box-shadow: inset 0 1px 1px rgba(0, 0, 0, 0.075);
  transition: border-color ease-in-out 0.15s, box-shadow ease-in-out 0.15s;
  &:focus {
    border-width: 2px !important;
    border-style: solid !important;
    border-color: rgb(92, 106, 196) !important;
    border-image: initial !important;
    outline: white;
  }
",
            ],
          },
          "displayName": "Styled(styled.input)",
          "foldedComponentIds": Array [
            "sc-bdVaJa",
          ],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "input",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-dnqmqq",
          "isStatic": false,
          "rules": Array [
            "
  ",
            [Function],
            "
",
          ],
        },
        "displayName": "Styled(styled.p)",
        "foldedComponentIds": Array [
          "sc-htoDjs",
        ],
        "render": [Function],
        "styledComponentId": "sc-dnqmqq",
        "target": "p",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
